### DISCUSSION TITLE
treeshakable without bundler for cdn consumers

### DISCUSSION THREAD

**Discussion by bwyx** [open-circle/valibot#126] [Ideas]
I use importmap without bundler, with current [build output](https://www.npmjs.com/package/valibot?activeTab=code) cdn consumers still have to download all code to use directly on the browser.

Splitting each method into individual files or paths should work.

**Comment by bwyx** [comment]
even though full size gzipped valibot is not even half the zod haha

**Comment by fabian-hiller** [comment]
If I did that, the bundle size would probably end up being the same, since almost no compression can be applied, since you would have to load each function individually. Do you have examples of other libraries?

**Comment by fabian-hiller** [comment]
What you can do is export what you need with bundlejs.com, run the build step and then copy the output your project: https://bundlejs.com/?q=valibot&treeshake=%5B%7B+email%2CminLength%2Cobject%2Cstring+%7D%5D
